frontend/src/views/admin/matching/sorts.tsx

```diff
@@ -0,0 +1,472 @@
+                        items = [...sortList];
+                        items[index] = {
+                            ...items[index],
+                            asc: !items[index]["asc"],
+                        };
+                        setSortList(items);
+                    }}
+                >
+                    {" "}
+                    {sortList[index]["asc"] ? (
+                        <TiArrowSortedUp />
+                    ) : (
+                        <TiArrowSortedDown />
+                    )}{" "}
+                </div>
+            )}
+            {selected && (
+                <div
+                    className="sort-icon"
+                    onClick={() => {
+                        // Button for removing this sort
+                        items = [...sortList];
+                        items.splice(index, 1);
+                        setSortList(items);
+                    }}
+                >
+                    <GrFormClose />
+                </div>
+            )}
+        </>
+    );
+}
+
+/**
+ * Applies a set of sorting functions outlined by "sortList" to a list of applicant summaries.
+ *
+ * @param {ApplicantSummary[]} applicantSummaries
+ * @param {sortMapItem[]} sortList
+ * @param {Position | null} position
+ * @returns
+ */
+export function applySorts(
+    applicantSummaries: ApplicantSummary[],
+    sortList: sortMapItem[],
+    position: Position | null
+) {
+    // Return early if any inputs aren't defined
+    if (!position || applicantSummaries.length === 0 || sortList.length === 0) {
+        return;
+    }
+
+    currPosition = position;
+
+    // Apply each sort in the opposite order they appear in sortList
+    let reversedList = [...sortList].reverse();
+    for (const sortItem of reversedList) {
+        sortItem["function"](applicantSummaries, sortItem["asc"]);
+    }
+}
+
+// Wrapper function for handling ascending vs. descending sorts
+function flipIfDescending(val: number, asc: boolean) {
+    return val * (asc ? 1 : -1);
+}
+
+// Sorting functions -- all of these do in-place sorting
+function sortByFirstName(applicantSummaries: ApplicantSummary[], asc = true) {
+    applicantSummaries.sort((a, b) => {
+        return `${a.applicant.first_name}, ${a.applicant.last_name}`.toLowerCase() <
+            `${b.applicant.first_name}, ${b.applicant.last_name}`.toLowerCase()
+            ? flipIfDescending(-1, asc)
+            : flipIfDescending(1, asc);
+    });
+}
+
+function sortByLastName(applicantSummaries: ApplicantSummary[], asc = true) {
+    applicantSummaries.sort((a, b) => {
+        return `${a.applicant.last_name}, ${a.applicant.first_name}`.toLowerCase() <
+            `${b.applicant.last_name}, ${b.applicant.first_name}`.toLowerCase()
+            ? flipIfDescending(-1, asc)
+            : flipIfDescending(1, asc);
+    });
+}
+
+function sortByProgram(applicantSummaries: ApplicantSummary[], asc = true) {
+    const priority = ["U", "PD", "MScAC", "M", "P"];
```

Comment:
This needs a type to ensure it's exhaustive. I believe you're missing `Other`, which shows up in a different file?